src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.cpp

```diff
@@ -0,0 +1,759 @@
+    return pulse;
+
+  real64 const pi = 2.0*acos( 0 );
+  real64 const lam = (f0*pi)*(f0*pi);
+  pulse = 2.0*lam*(2.0*lam*(time_n-o_tpeak)*(time_n-o_tpeak)-1.0)*exp( -lam*(time_n-o_tpeak)*(time_n-o_tpeak));
+
+  return pulse;
+}
+
+
+
+real64 AcousticWaveEquationSEM::explicitStep( real64 const & time_n,
+                                              real64 const & dt,
+                                              integer const cycleNumber,
+                                              DomainPartition & domain )
+{
+
+  GEOSX_UNUSED_VAR( time_n, dt, cycleNumber );
+
+  MeshLevel & mesh = domain.getMeshBody( 0 ).getMeshLevel( 0 );
+
+  NodeManager & nodeManager = mesh.getNodeManager();
+
+  arrayView1d< real64 const > const mass = nodeManager.getExtrinsicData< extrinsicMeshData::MassVector >();
+  arrayView1d< real64 const > const damping = nodeManager.getExtrinsicData< extrinsicMeshData::DampingVector >();
+
+  arrayView1d< real64 > const p_nm1 = nodeManager.getExtrinsicData< extrinsicMeshData::Pressure_nm1 >();
+  arrayView1d< real64 > const p_n = nodeManager.getExtrinsicData< extrinsicMeshData::Pressure_n >();
+  arrayView1d< real64 > const p_np1 = nodeManager.getExtrinsicData< extrinsicMeshData::Pressure_np1 >();
+
+  /// get array of indicators: 1 if node on free surface; 0 otherwise
+  arrayView1d< localIndex const > const freeSurfaceNodeIndicator = nodeManager.getExtrinsicData< extrinsicMeshData::FreeSurfaceNodeIndicator >();
+
+  arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const X = nodeManager.referencePosition().toViewConst();
+
+  arrayView1d< real64 > const stiffnessVector = nodeManager.getExtrinsicData< extrinsicMeshData::StiffnessVector >();
+
+  arrayView1d< real64 > const rhs = nodeManager.getExtrinsicData< extrinsicMeshData::ForcingRHS >();
+
+  forTargetRegionsComplete( mesh, [&]( localIndex const,
+                                       localIndex const,
+                                       ElementRegionBase & elemRegion )
+  {
+    elemRegion.forElementSubRegionsIndex< CellElementSubRegion >( [&]( localIndex const,
+                                                                       CellElementSubRegion & elementSubRegion )
+    {
+      arrayView2d< localIndex const, cells::NODE_MAP_USD > const & elemsToNodes = elementSubRegion.nodeList();
+
+      finiteElement::FiniteElementBase const &
+      fe = elementSubRegion.getReference< finiteElement::FiniteElementBase >( getDiscretizationName() );
+      finiteElement::dispatch3D( fe,
+                                 [&]
+                                   ( auto const finiteElement )
+      {
+        using FE_TYPE = TYPEOFREF( finiteElement );
+
+        constexpr localIndex numNodesPerElem = FE_TYPE::numNodes;
+        constexpr localIndex numQuadraturePointsPerElem = FE_TYPE::numQuadraturePoints;
+
+        real64 N[numNodesPerElem];
+        real64 gradN[ numNodesPerElem ][ 3 ];
+
+        for( localIndex k=0; k < elemsToNodes.size( 0 ); ++k )
```

Comment:
`forAll< serialPolicy >`

Reply:
For this, I have local arrays that we need redefine to be compliant with the Raja memory policies and I did not define the kernel yet that is why I keep using the simple loop for now.  
I wanted make this first version available in develop before introducing the Raja loops and start testing on different platform.